dandi/tests/test_models.py

```diff
@@ -154,3 +166,265 @@ def test_autogenerated_titles():
     assert schema["title"] == "Asset Meta"
     assert schema["properties"]["schemaVersion"]["title"] == "Schema Version"
     assert schema["definitions"]["PropertyValue"]["title"] == "Property Value"
+
+
+def datacite_post(datacite, doi):
+    """
+    posting the datacite object and returning the status codes
+    of requests.post and requests.get
+    """
+
+    # removing doi in case it exists
+    _clean_doi(doi)
+
+    # checking f I'm able to create doi
+    rp = requests.post(
+        "https://api.test.datacite.org/dois",
+        json=datacite,
+        headers={"Content-Type": "application/vnd.api+json"},
+        auth=("DARTLIB.DANDI", os.environ["DATACITE_DEV_PASSWORD"]),
+    )
+
+    # checking if i'm able to get the url
+    rg = requests.get(url=f"https://api.test.datacite.org/dois/{doi}/activities")
+
+    # cleaning url
+    _clean_doi(doi)
+    return rp.status_code, rg.status_code
+
+
+def _clean_doi(doi):
+    """removing doi, ignoring the status code"""
+    requests.delete(
+        f"https://api.test.datacite.org/dois/{doi}",
+        auth=("DARTLIB.DANDI", os.environ["DATACITE_DEV_PASSWORD"]),
+    )
+
+
+@pytest.fixture(scope="module")
+def schema():
+    sr = requests.get(
+        "https://raw.githubusercontent.com/datacite/schema/master/source/"
+        "json/kernel-4.3/datacite_4.3_schema.json"
+    )
+    schema = sr.json()
+    return schema
+
+
+@pytest.fixture(scope="module")
+def client():
+    return DandiAPIClient("https://api.dandiarchive.org/api/")
```

Comment:
I think we could use the one we have in the fixture here without relying on the live one and existence of specific dandisets there.

only 000027 (well and 000027) is the dandiset we have under "our control" and the ones which could be used in the tests.

if tests talk to real server, I think we should annotate them somehow, probably with `@pytest.mark.integration`.

@jwodder please review and provide comments and suggestions to get this PR landed.

Reply:
I also could come back to the previous version where I used metadata from real dandiset as examples, but I saved the metadata as a json file in the repo

Reply:
yeah -- we should not rely on testing on real arbitrary datasets out of our control, at least until we know that they are metadata records are as kosher as they could be and that we have safeguards that nobody could make them invalid ;)   but ok, let's proceed as is for now and adjust later.

Reply:
this has been removed after our discussion on Monday

Reply:
Thank you @djarecka ! I must had been looking at the outdated diff while checking on this before merge :-/ Great!